analyses/oncoprint-landscape/01-plot-oncoprint.R

```diff
@@ -251,6 +239,37 @@ maf_object <- prepare_maf_object(
   fusion_df = fusion_df
 )
 
+#### Subset MAF Object (Optional)----------------------------------------------
+
+# Code here is specifically adapted from:
+# https://github.com/marislab/create-pptc-pdx-oncoprints/blob/master/R/create-complexheat-oncoprint-revision.R
+
+# Subset `maf_object` for histology-specific goi list
+if (!is.null(opt$goi_list)) {
+  maf_object = subsetMaf(
+    maf = maf_object,
+    tsb = metadata$Tumor_Sample_Barcode,
+    genes = goi_list,
+    mafObj = TRUE
+  )
+  
+  # Get top mutated genes per this subset object
+  gene_sum <- mafSummary(maf_object)$gene.summary
+  
+  # Sort to get top altered genes rather than mutated only genes
+  goi_ordered <-
+    gene_sum[order(gene_sum$AlteredSamples, decreasing = T),]
+  
+  if (!is.null(opt$top_n)) {
+    
+    # Select top `n` genes if the argument is provided
+    top_n <- ifelse(nrow(gene_sum) < opt$top_n, nrow(gene_sum), opt$top_n)
+    
+    goi_list <- goi_ordered[1:top_n,]
+    
+  }
+  
+}
```

Comment:
The actual code changes I'm suggesting are untested, but I believe you will only have to take these steps if someone specifies a GOI list _and_ a top n, so you can simplify this a bit:

```suggestion
# Subset `maf_object` for histology-specific goi list
if (!is.null(opt$goi_list) & !is.null(opt$top_n)) {
  
  maf_object <- subsetMaf(
    maf = maf_object,
    tsb = metadata$Tumor_Sample_Barcode,
    genes = goi_list,
    mafObj = TRUE
  )
  
  # Get top mutated genes per this subset object
  gene_sum <- mafSummary(maf_object)$gene.summary
  
  # Sort to get top altered genes rather than mutated only genes
  goi_ordered <-
    gene_sum[order(gene_sum$AlteredSamples, decreasing = T),]
  
  # Select top `n` genes if the argument is provided
  top_n <- ifelse(nrow(gene_sum) < opt$top_n, nrow(gene_sum), opt$top_n)
  goi_list <- goi_ordered[1:top_n,]
  
}
```

Reply:
Upon testing this method does not appear to obey the `top_n` argument.

Reply:
Can you expand on that a bit?

Reply:
It does not limit the amount of genes being displayed to `top_n`, it instead shows all of the genes listed in the goi list.

Reply:
But the two `if` way worked?

Reply:
That's correct.

Reply:
That suggests that the logic _isn't quite right_ because it's using the original list (`goi_list`) - the conditions for the `if()` are not being met. Since you don't want to subset the MAF and do the `mafSummary()` steps unless you have to and you only have to if you have a `top_n` argument I would see if you can get to the bottom of it.